File: src/features/education/domain/EducationCourse.js
import { createEntity } from '../../../shared/domain/createEntity';

const schema = {
  courseId: { label: '교육과정 ID', required: true },
  courseName: { label: '과정명', required: true },
  educationType: { label: '교육 유형', required: true },
  defaultHours: { label: '기본 교육시간', type: 'number', min: 1 },
};

export function createEducationCourse(input) {
  return createEntity(input, schema);
}

File: backend/src/main/java/com/oose/labsafety/education/domain/EducationCourse.java
package com.oose.labsafety.education.domain;

import com.oose.labsafety.common.domain.Identifiable;

public record EducationCourse(
        String courseId,
        String courseName,
        String educationType,
        int defaultHours
) implements Identifiable {
    @Override
    public String id() {
        return courseId;
    }
}
